File: python-service/functions/list.py
from nitric.api.exception import NitricServiceException
from nitric.faas import start, Trigger, Response
from nitric.api import Documents


async def handler(trigger: Trigger) -> Response:
    response = trigger.default_response()

    try:
        examples_query = Documents().collection("example").query()
        response.data = [doc.content async for doc in examples_query.stream()]
    except NitricServiceException:
        response.context.as_http().status = 500
        response.data = "An unexpected error occurred"

    return response


if __name__ == "__main__":
    start(handler)

File: java-stack/src/main/java/function/List.java
package function;

import common.Example;
import com.fasterxml.jackson.databind.ObjectMapper;

import io.nitric.api.document.Documents;
import io.nitric.api.document.ResultDoc;
import io.nitric.faas.Faas;

import java.io.IOException;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class List {

    public static void main(String[] args) {
        new Faas()
            .http(context -> {
                Stream<ResultDoc<Example>> stream = new Documents().collection("examples")
                    .query(Example.class)
                    .stream();
        
                java.util.List<Example> examples = stream
                    .map(ResultDoc::getContent)
                    .collect(Collectors.toUnmodifiableList());
        
                try {
                    var json = new ObjectMapper().writeValueAsString(examples);
        
                    context.getResponse()
                        .contentType("application/json")
                        .data(json);
        
                } catch (IOException ioe) {
                    context.getResponse()
                        .status(500)
                        .data("Error querying examples: " + ioe.toString());
                }
        
                return context;
            })
            .start();
    }

}